pkg/ddevapp/drupal.go

```diff
@@ -522,3 +622,55 @@ func createDrupal8SyncDir(app *DdevApp) error {
 
 	return nil
 }
+
+// settingsHasInclude determines if the settings.php or equivalent includes settings.ddev.php or equivalent.
+// This is done by looking for the ddev settings file (settings.ddev.php) in settings.php.
+func settingsHasInclude(drupalConfig *DrupalSettings, siteSettingsPath string) (bool, error) {
+	included, err := fileutil.FgrepStringInFile(siteSettingsPath, drupalConfig.SiteSettingsLocal)
+	if err != nil {
+		return false, err
+	}
+
+	return included, nil
+}
+
+// addIncludeToDrupalSettingsFile modifies the settings.php file to include the settings.ddev.php
+// file, which contains ddev-specific configuration.
+func addIncludeToDrupalSettingsFile(drupalConfig *DrupalSettings, siteSettingsPath string) error {
+	// Open file for appending to preserve current contents
+	file, err := os.OpenFile(siteSettingsPath, os.O_RDWR|os.O_APPEND, 0644)
+	if err != nil {
+		return err
+	}
+	defer util.CheckClose(file)
+
+	// Get current contents of settings.php
+	currentSiteSettings, err := ioutil.ReadAll(file)
+	if err != nil {
+		return err
+	}
+
+	// If settings.php is empty, write the complete settings template
+	if len(currentSiteSettings) == 0 {
+		if err = writeDrupalCommonSettingsFile(drupalConfig, siteSettingsPath); err != nil {
+			return err
+		}
+	}
+
+	// The file is not empty, append the include
+	tmpl, err := template.New("settings").Funcs(getTemplateFuncMap()).Parse(drupalCommonSettingsAppendTemplate)
+	if err != nil {
+		return err
+	}
+
+	b := bytes.NewBuffer([]byte{})
+	if err := tmpl.Execute(b, drupalConfig); err != nil {
+		return err
+	}
+
+	if _, err := file.Write(b.Bytes()); err != nil {
+		return err
+	}
```

Comment:
Silly me, I had envisioned just an append to the file using perhaps unix tools, but that certainly wouldn't work. But wouldn't `f, err := os.OpenFile(filename, os.O_APPEND|os.O_WRONLY, 0644)` work?

I do understand that we might be creating as opposed to appending, but seems like a large-hammer approach to appending.

Reply:
This got a little logically polluted from the an earlier revision of this function that used file.WriteAt().  There were a couple refactor/flow improvement opportunities here that I took advantage of, it should now be clearer.